File: Data Structures & Algorithms/binary-search/submission-11.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size() - 1;
        int m = nums.size() / 2;
        while (nums[m] != target && l < r) {
            if (target < nums[m]) {
                r = m - 1;
                m = (l + r) / 2;
            } else if (target > nums[m]) {
                l = m + 1;
                m = (l + r) / 2;
            }
        }
        if (nums[m] == target) {
            return m;
        }
        return -1;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-11.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = defaultdict(int)

        for i in range(len(nums)):
            difference = target - nums[i]
            if difference in seen.keys() and seen[difference] != i:
                return [seen[difference], i]
            seen[nums[i]] = i